agent/lib/kontena/workers/node_info_worker.rb

```diff
@@ -236,6 +253,54 @@ def calculate_container_time(container)
       0
     end
 
+    # @param [Array<Vmstat::Cpu>] prev_cpu_stats
+    # @param [Array<Vmstat::Cpu>] current_cpu_stats
+    # @return [Hash]
+    def calculate_average_cpu(prev_cpu_stats, current_cpu_stats)
+      all = prev_cpu_stats.zip(current_cpu_stats).map do |prev, current|
+        system_ticks = current.system - prev.system
+        user_ticks = current.user - prev.user
+        idle_ticks = current.idle - prev.idle
+
+        total_ticks = system_ticks + user_ticks + idle_ticks
+
+        [system_ticks, user_ticks, idle_ticks].map do |ticks|
+          (ticks / total_ticks.to_f) * 100.0
+        end
+      end
+
+      averages = all.transpose.map do |stats|
+        total = stats.reduce :+
+        total / stats.size.to_f
+      end
+
+      return {
+        system: averages[0],
+        user: averages[1],
+        idle: averages[2]
+      }
+    end
+
+    # @param [Vmstat::NewtworkInterface] previous_iface
+    # @param [Vmstat::NetworkInterface] current_iface
+    # @param [Number] interval
+    # @return [Hash]
+    def calculate_network_traffic(previous_iface, current_iface, interval)
+      in_bytes_per_second = (current_iface.in_bytes - previous_iface.in_bytes) / interval
+      out_bytes_per_second = (current_iface.out_bytes - previous_iface.out_bytes) / interval
+
+      return {
+        in_bytes_per_second: in_bytes_per_second,
+        out_bytes_per_second: out_bytes_per_second
+      }
+    end
+
+    def get_network_interface()
```

Comment:
@SpComb @nevalla : Any thoughts on this algorithm for getting the network interface to monitor?  I'm just grabbing the ethernet adapter with the most outgoing traffic.  Wasn't sure if I should instead get a sum of all ethernet (or non-loopback) adapters I/O.